app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -199,23 +200,26 @@ internal fun NiaApp(
                     )
                 }
 
-                NiaNavHost(
-                    appState = appState,
-                    onShowSnackbar = { message, action ->
-                        snackbarHostState.showSnackbar(
-                            message = message,
-                            actionLabel = action,
-                            duration = Short,
-                        ) == ActionPerformed
-                    },
+                Box(
                     modifier = if (shouldShowTopAppBar) {
```

Comment:
This should be equivalent: applying a `Modifier` directly or wrapping it in a `Box` with the `Modifier` applied. For some reason though, the old way would cause the `currentCompositeKeyHash` to change in the inner code, and therefore wipe away saved instance state.

Reply:
Very strange. Thanks for investigating and implementing this workaround.